tiledb/sm/filesystem/azure.cc

```diff
@@ -671,23 +562,21 @@ Status Azure::copy_blob(const URI& old_uri, const URI& new_uri) {
   std::string old_container_name;
   std::string old_blob_path;
   RETURN_NOT_OK(parse_azure_uri(old_uri, &old_container_name, &old_blob_path));
+  std::string source_uri = client_->GetBlobContainerClient(old_container_name)
+                               .GetBlobClient(old_blob_path)
+                               .GetUrl();
 
   std::string new_container_name;
   std::string new_blob_path;
   RETURN_NOT_OK(parse_azure_uri(new_uri, &new_container_name, &new_blob_path));
 
-  std::future<azure::storage_lite::storage_outcome<void>> result =
-      client_->start_copy(
-          old_container_name, old_blob_path, new_container_name, new_blob_path);
-  if (!result.valid()) {
+  try {
+    client_->GetBlobContainerClient(new_container_name)
+        .GetBlobClient(new_blob_path)
+        .StartCopyFromUri(source_uri);
```

Comment:
Note to myself for next week: `StartCopyFromUri`'s return type has a [`PollUntilDone`](https://azuresdkdocs.blob.core.windows.net/$web/cpp/azure-core/1.8.1/class_azure_1_1_core_1_1_operation.html#a8486143c77f4f57b4038f12df803cba4) method that we can use instead of `wait_for_blob_to_propagate`.